lib/topological_inventory/sync/inventory_upload_validator_worker.rb

```diff
@@ -32,9 +32,15 @@ def perform(message)
 
       private
 
-      def valid_payload?(_payload)
-        # TODO validate that this is a correct topological inventory payload
-        true
+      def valid_payload?(payload)
+        schema_name = payload.dig("metadata", "schema", "name")
+        schema_klass = schema_klass_name(schema_name).safe_constantize
+
+        schema_klass.present?
+      end
+
+      def schema_klass_name(name)
```

Comment:
We do this exact same thing in the topological_inventory-persister...how about adding a module method to `TopologicalInventory::Schema` to load the appropriate schema?

Reply:
Something like `TopologicalInventory::Schema.klass_for(name)`

Reply:
good idea :+1: